server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/DomainRoutes.java

```diff
@@ -59,17 +77,73 @@ public DomainRoutes(DomainList domainList, JsonTransformer jsonTransformer) {
 
     @Override
     public void define(Service service) {
-        service.get(DOMAINS,
-            (request, response) -> domainList.getDomains(),
-            jsonTransformer);
+        this.service = service;
 
-        service.get(SPECIFIC_DOMAIN, this::exists);
+        getDomains();
 
-        service.put(SPECIFIC_DOMAIN, this::addDomain);
+        domainIsExisted();
+
+        addDomain();
 
+        deleteDomain();
```

Comment:
No, it's not only definition, it's real function and action so that name is still ok I think